helix-core/src/main/java/org/apache/helix/model/IdealState.java

```diff
@@ -55,7 +55,9 @@
     MAX_PARTITIONS_PER_INSTANCE,
     INSTANCE_GROUP_TAG,
     REBALANCER_CLASS_NAME,
-    HELIX_ENABLED
+    HELIX_ENABLED,
+    RESOURCE_GROUP_NAME,
+    RESOURCE_GROUP_ENABLED
```

Comment:
ResourceGroupEnabled may be a little confusing, I will change it to GroupRoutingEnabled. This means routingProvider should consider aggregate all partition/instances from all resources from the same resourceGroup.  ResourceGroup Name and Instance Tag for a resource can be used for other purpose, only when the GroupRoutingEnabled is true, they are combined together to provide this group routing feature.  This also says, GroupRoutingEnabled is true implicitly means ResourceGroup and ResourceTag (instanceTag) are set in the IdealState, and resourceTag is identical in each resource from the same resource group.   But this constraints are not necessary if GroupRoutingEnable is false.  For example, now many resources could have same instanceGroupTag.